midend/predication.cpp

```diff
@@ -28,13 +175,68 @@ const IR::Expression* Predication::clone(const IR::Expression* expression) {
     return expression->apply(cloner);
 }
 
-const IR::Node* Predication::postorder(IR::AssignmentStatement* statement) {
-    if (!inside_action || ifNestingLevel == 0)
+const IR::AssignmentStatement* Predication::clone(const IR::AssignmentStatement* statement) {
+    // We often need to clone assignments.  This is necessary because
+    // in the end we will generate different code for the different clones of
+    // an assignments.
+    ClonePathExpressions cloner;
+    auto clonedAsssignment = new IR::AssignmentStatement(*statement);
+    clonedAsssignment->left->apply(cloner);
+    clonedAsssignment->right->apply(cloner);
+    return clonedAsssignment;
```

Comment:
seems this could be just

    return statement->apply(ClonePathExpressions());